### PR TITLE
[proxy] Upgrade Caddy

### PR DESCRIPTION
**Pull request by aledbf**
## Description

Upgrade Caddy to latest stable version

## Related Issue(s)

Upgrade required after rebasing https://github.com/gitpod-io/gitpod/pull/5468 and getting a compilation error 
https://werft.gitpod-dev.com/job/gitpod-build-aledbf-typescript44.3

## How to test

Logs of the `proxy` pod should not contain errors

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankoehnlein** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankoehnlein` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5615#issuecomment-915704374" title="Approved">aledbf</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankoehnlein** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankoehnlein` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
@csweichel: GitHub didn't allow me to assign the following users: engineering-meta.

Note that only [gitpod-io members](https://github.com/orgs/gitpod-io/people), repo collaborators and people who have commented on this issue/PR can be assigned. Additionally, issues/PRs can only have 10 assignees at the same time.
For more information please see [the contributor guide](https://git.k8s.io/community/contributors/guide/first-contribution.md#issue-assignment-in-github)

<details>

In response to [this](https://github.com/gitpod-io/gitpod/pull/5615#issuecomment-915786130):

>/assign engineering-meta


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by csweichel** [comment]
Proxy is owned by the meta team - hence the original reviewers are in a good position to review :)

File: components/proxy/Dockerfile
Comment (geropl) [inline components/proxy/Dockerfile:15]: :tada:

File: components/proxy/plugins/corsorigin/go.mod
Comment (geropl) [inline components/proxy/plugins/corsorigin/go.mod:10]: Why does materializing all references instead of just `require github.com/caddyserver/caddy/v2 v2.4.3` work now? Because we're not no longer on a fork? Just trying to understand. :-)

**Review by geropl — APPROVED** [review]
:+1:

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 769812f000efa81d08c4ca6c141b27ade6cdeb3e</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5615#issuecomment-915704374" title="Approved">aledbf</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5615#pullrequestreview-750013319" title="Approved">geropl</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5615#issuecomment-915704374" title="Approved">aledbf</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5615#pullrequestreview-750013319" title="Approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5615#issuecomment-915913301" title="Approved">JanKoehnlein</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)~~ [JanKoehnlein,geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5615#issuecomment-915704374" title="Approved">aledbf</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5615#pullrequestreview-750013319" title="Approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5615#issuecomment-915913301" title="Approved">JanKoehnlein</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/5615#issuecomment-915913301" title="Approved">JanKoehnlein</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)~~ [JanKoehnlein,geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/proxy/plugins/corsorigin/go.mod
Comment (aledbf) [reply components/proxy/plugins/corsorigin/go.mod:10]: This is a change in [Go 1.17](https://golang.org/doc/go1.17#go-command) :(

**Pull request merged**
